lib/bundler/dependency.rb

```diff
@@ -74,15 +74,6 @@ class Dependency < Gem::Dependency
       :x64_mingw_26 => Gem::Platform::X64_MINGW,
     }.freeze
 
-    REVERSE_PLATFORM_MAP = {}.tap do |reverse_platform_map|
-      PLATFORM_MAP.each do |key, value|
-        reverse_platform_map[value] ||= []
-        reverse_platform_map[value] << key
-      end
-
-      reverse_platform_map.each {|_, platforms| platforms.freeze }
-    end.freeze
```

Comment:
This is a part of reverting 0a8ca4879e0b79aa4109a0dc424940b079ef38d0.
See also: #4150 and #4102.